### PR TITLE
refactor(Apps/Defines): fix codefactor warning

### PR DESCRIPTION
**Pull request by Kitzunu**
According to Codefactor 
(https://www.codefactor.io/repository/github/azerothcore/azerothcore-wotlk/file/master/apps/bash_shared/defines.sh)

using backticks is deprecated and should be replaced with $()

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: apps/compiler/includes/functions.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ function comp_configure() {
 
 
 function comp_compile() {
-  [ $MTHREADS == 0 ] && MTHREADS=`grep -c ^processor /proc/cpuinfo` && MTHREADS=$(($MTHREADS + 2))
+  [ $MTHREADS == 0 ] && MTHREADS=$(grep -c ^processor /proc/cpuinfo) && MTHREADS=$(($MTHREADS + 2))
```
Comment (Helias) [inline apps/compiler/includes/functions.sh:49]: not sure if are needed the double quote outside $(grep...) or in $(git...)

Comment (Kitzunu) [reply apps/compiler/includes/functions.sh:49]: ?

**Review by FrancescoBorzi — APPROVED** [review]

File: apps/compiler/includes/functions.sh (hunk 1 above)
Comment (Helias) [reply apps/compiler/includes/functions.sh:49]: example
```bash
_cur_branch="$(git --git-dir="$J_PATH/.git/" --work-tree="$J_PATH/" rev-parse --abbrev-ref HEAD)"
```

or something similar

Comment (Kitzunu) [reply apps/compiler/includes/functions.sh:49]: Not according to codefactor, but I never programmed in whatever language this is

**Pull request merged**